src/components/Chart.js

```diff
@@ -24,9 +23,9 @@ const Chart = ({ stock, latestPrice, updateChartRange }) => {
 
     return (
         <div className="chart">
-            {stock.length === 0
-                ? <div className="loading-spinner"></div>
-                : <>
+        {(stock === null) ? <div className="loading-spinner"></div> :
```

Comment:
nit: Take care in formatting here, as this can be awkward to read when a ternary is split over 2 lines in an odd way.